### PR TITLE
Bump OSC version to 1.8.0, and document where version appears

### LINKED ISSUE #465: Bump version number of the operator to 1.8.0
The development branch of the repository should refer to the upcoming release.

### PR DESCRIPTION
**Pull request by c3d**
Bump the version number of OSC to the upcoming release, 1.8.0, and document where the version number appears in the sources.

When starting a new version, many locations need to be updated (here shown with current version `1.5.2` on `main` branch):

```
Makefile:6:VERSION ?= 1.5.2
config/manager/kustomization.yaml:16:  newTag: 1.5.2
config/manifests/bases/sandboxed-containers-operator.clusterserviceversion.yaml:16:    olm.skipRange: '>=1.1.0 <1.5.2'
config/manifests/bases/sandboxed-containers-operator.clusterserviceversion.yaml:28:  name: sandboxed-containers-operator.v1.5.2
config/manifests/bases/sandboxed-containers-operator.clusterserviceversion.yaml:368:  version: 1.5.2
config/samples/deploy.yaml:9: image:  quay.io/openshift_sandboxed_containers/openshift-sandboxed-containers-operator-catalog:v1.5.2
config/samples/deploy.yaml:39:  startingCSV: sandboxed-containers-operator.v1.5.2
hack/aws-image-job.yaml:24:        image: registry.redhat.io/openshift-sandboxed-containers/osc-podvm-payload-rhel9:1.5.2
hack/azure-image-job.yaml:23:        image: registry.redhat.io/openshift-sandboxed-containers/osc-podvm-payload-rhel9:1.5.2
```

The places needing change above have been labeled with `OSC_VERSION`. The reason is that version number `1.5.2` itself appears in places that are not related to OSC, for example:

```
go.sum:966:github.com/golang/protobuf v1.5.2/go.mod h1:XVQd3VNwM+JqD3oG2Ue2ip4fOMUkwXdXDdiuN0vRsmY=
go.sum:1760:github.com/rogpeppe/go-internal v1.5.2/go.mod h1:xXDCJY+GAPziupqXw64V24skbSoqbTEfhy4qGm1nDQc=
go.sum:1768:github.com/russross/blackfriday v1.5.2/go.mod h1:JO/DiYxRf+HjHt06OyowR9PTA263kcR/rfWxYHBV53g=
```

As we are migrating towards 1.8.0, there are even more places referring to a 1.8.0 version that is not related to OSC:

```
go.sum:43:cloud.google.com/go/bigquery v1.8.0/go.mod h1:J5hqkt3O0uAFnINi6JXValWIb1v0goeZM77hZzJN/fQ=
go.sum:66:cloud.google.com/go/storage v1.8.0/go.mod h1:Wv1Oy7z6Yz3DshWRJFhqM/UCfaWIRTdp0RXyy7KQOVs=
go.sum:837:github.com/go-stack/stack v1.8.0/go.mod h1:v0f6uXyyMGvRgIKkXu+yp6POWl0qKG85gN/melR3HDY=
go.sum:1092:github.com/gorilla/mux v1.8.0/go.mod h1:DVbg23sWSpFRCP0SfiEN6jmj59UnW/n46BH5rLB71So=
go.sum:1293:github.com/karrick/godirwalk v1.8.0/go.mod h1:H5KPZjojv4lE+QYImBI8xVtrBRgYrIVsaRPx4tDPEn4=
go.sum:1361:github.com/lib/pq v1.8.0/go.mod h1:AlVN5x4E4T544tWzH6hKfbfQvm3HdbOxrmggDNAPY9o=
go.sum:1373:github.com/magiconair/properties v1.8.0/go.mod h1:PppfXfuXeibc/6YijjN8zIbojt8czPbwD3XqdrwzmxQ=
go.sum:1557:github.com/onsi/ginkgo v1.8.0/go.mod h1:lLunBs/Ym6LB5Z9jYTR76FiuTmxDTDusOGeTQH+WWjE=
go.sum:1629:github.com/opencontainers/selinux v1.8.0/go.mod h1:RScLhm78qiWa2gbVCcGkC7tCGdgk3ogry1nUQF8Evvo=
go.sum:1890:github.com/stretchr/testify v1.8.0/go.mod h1:yNjHg4UonilssWZ8iaSj1OCr/vHnekPRkoO+kdMU+MU=
go.sum:2828:google.golang.org/grpc v1.8.0/go.mod h1:yo6s7OP7yaDglbqo1J04qKzAhqBH6lvTonzMVmEdcZw=
```

**- Description of the problem which is fixed/What is the use case**

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by c3d** [comment]
@gkurz Not sure why the tests fail? I'm not familiar with this test suite, and the logs don't speak to me.

**Comment by gkurz** [comment]
Hi @c3d !

> When starting a new version, many locations need to be updated (here shown with current version 1.5.2 on main branch):

FYI we don't use the `main` branch anymore. Only 'devel' and `release-X.Y` branches.

> Not sure why the tests fail? I'm not familiar with this test suite, and the logs don't speak to me.

`ci/prow/check` fails because `make test` is broken since #300. The test isn't required and can be safely ignored for now.

`ci/prow/sandboxed-containers-operator-e2e` has been broken forever AFAIK. It isn't required either and can be ignored as well.

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -13,4 +13,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: quay.io/openshift_sandboxed_containers/openshift-sandboxed-containers-operator
-  newTag: v1.7.0
+  newTag: 1.8.0  ## OSC_VERSION
```
Comment (gkurz) [inline config/manager/kustomization.yaml]:
```suggestion
  newTag: v1.8.0  ## OSC_VERSION
```
Please keep the `v` here to be consistent with Operator SDK.

File: docs/DEVELOPMENT.md [hunk 2]
```diff
@@ -131,5 +131,42 @@ Uncomment all entries marked with `[CERTMANAGER]` in manifest files under `confi
...
+hack/azure-image-job.yaml:23:        image: registry.redhat.io/openshift-sandboxed-containers/osc-podvm-payload-rhel9:1.5.2
+```
+
+These places have been labeled with `OSC_VERSION`. The reason is that version number `1.5.2` appears in places that are not related to OSC
+
+```
+go.sum:966:github.com/golang/protobuf v1.5.2/go.mod h1:XVQd3VNwM+JqD3oG2Ue2ip4fOMUkwXdXDdiuN0vRsmY=
+go.sum:1760:github.com/rogpeppe/go-internal v1.5.2/go.mod h1:xXDCJY+GAPziupqXw64V24skbSoqbTEfhy4qGm1nDQc=
+go.sum:1768:github.com/russross/blackfriday v1.5.2/go.mod h1:JO/DiYxRf+HjHt06OyowR9PTA263kcR/rfWxYHBV53g=
+```
```
Comment (gkurz) [inline docs/DEVELOPMENT.md]: Maybe just mention that `go.mod` or `go.sum` should be ignored as we don't have anything that would bear the actual version of OSC in there ?

**Review by gkurz — COMMENTED** [review]
All the version changes LGTM and I like the OSC_VERSION tags that will ease future updates.

I've added some comments. PTAL.

**Comment by gkurz** [comment]
@c3d thanks to #464 and https://github.com/openshift/release/pull/57546 , you should get an all green CI if you rebase :smile:

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (c3d) [reply config/manager/kustomization.yaml]: Good catch, @gkurz. Thanks.

File: docs/DEVELOPMENT.md (hunk 2 above)
Comment (c3d) [reply docs/DEVELOPMENT.md]: I reworded that section.

**Comment by c3d** [comment]
> @c3d thanks to #464 and [openshift/release#57546](https://github.com/openshift/release/pull/57546) , you should get an all green CI if you rebase 😄

Rebased and took your other input into account, @gkurz. Please take another look.

**Review by gkurz — APPROVED** [review]
/lgtm

Thanks @c3d !

**Review by bpradipt — APPROVED** [review]
/lgtm
@c3d nice work on adding the OSC_VERSION tag. Will make it easier when updating the version

**Pull request merged**